internal/storage/image.go

```diff
@@ -589,9 +587,8 @@ type pullImageArgs struct {
 }
 
 type pullImageOutputItem struct {
-	Progress     *types.ProgressProperties `json:",omitempty"`
-	Result       string                    `json:",omitempty"` // If not "", in the format of transport.ImageName()
-	CanonicalRef string                    `json:",omitempty"` // If not "", in the format of RegistryImageReference.StringForOutOfProcessConsumptionOnly(), and always contains a digest.
+	Progress *types.ProgressProperties `json:",omitempty"`
+	Result   string                    `json:",omitempty"` // If not "", in the format of RegistryImageReference.StringForOutOfProcessConsumptionOnly(), and always contains a digest.
```

Comment:
I find myself wondering if we should keep calling this CanonicalRef so we don't have confusions between 1.30 and 1.31 in making this change (same variable name, same type, different semantic meaning)

Reply:
I did wonder about this — it is used via `reexec.Command`, i.e. talking to `/proc/self/exe`; i.e. there should never be a version drift between the producer’s and consumer’s implementation.

(And, also, `types.ImageReference` and `RegistryImageReference` string formats are mutually incompatible, so the consumer would fail parsing a mismatching value either way.)

---

I don’t care about the name of the field at all; it’s purely internal between `pullImageChild`/`pullImageParent`. Let me know what you want it to be.